torchgeo/trainers/byol.py

```diff
@@ -387,8 +387,16 @@ def training_step(self, *args: Any, **kwargs: Any) -> Tensor:
         """
         batch = args[0]
         x = batch["image"]
+
+        in_channels = self.hyperparams["in_channels"]
+        assert x.size(1) == in_channels or x.size(1) == 2 * in_channels
+        if x.size(1) == in_channels:
```

Comment:
This trainer now supports both datasets with and without RandomSeasonContrast

Reply:
Nice